src/crypto/algorithms/megolm.js

```diff
@@ -433,6 +472,9 @@ MegolmEncryption.prototype._getDevicesInRoom = function(room) {
     // have a list of the user's devices, then we already share an e2e room
     // with them, which means that they will have announced any new devices via
     // an m.new_device.
+    //
+    // XXX: what if the cache is stale, and the user left the room we had in common
+    // and then added new devices before joining this one? --Matthew
```

Comment:
yup, this is exactly what https://github.com/vector-im/riot-web/issues/2305#issue-178058560 is on about